### PR TITLE
Fixed load PSD files 

### LINKED ISSUE #1486: Thumbor could not load PSD files.
### Thumbor request URL
Call thumbor if this image (PSD file zipped) [guido.zip](https://github.com/thumbor/thumbor/files/9687141/guido.zip)

### Expected behaviour
Thumbor could load PSD file and convert to webp.

### Actual behaviour

Thumbor raise a error, because `engine/pil.py:create_image()` is trying to load `JpegImagePlugin.get_sampling(img)` on a `PsdImagePlugin.PsdImageFile`.

And PSD files does not contains attribute `layer`, but contains `layers`, passing condition added on this PR  #1480

```python
2022-09-30 19:01:48 thumbor:ERROR ERROR: Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/thumbor/handlers/__init__.py", line 212, in get_image
    result = await self._fetch(self.context.request.image_url)
  File "/usr/local/lib/python3.9/dist-packages/thumbor/handlers/__init__.py", line 876, in _fetch
    raise fetch_result.exception
  File "/usr/local/lib/python3.9/dist-packages/thumbor/handlers/__init__.py", line 844, in _fetch
    self.context.request.engine.load(fetch_result.buffer, extension)
  File "/usr/local/lib/python3.9/dist-packages/thumbor/engines/__init__.py", line 195, in load
    image_or_frames = self.create_image(buffer)
  File "/usr/local/lib/python3.9/dist-packages/thumbor/engines/pil.py", line 85, in create_image
    self.subsampling = JpegImagePlugin.get_sampling(img)
  File "/usr/local/lib/python3.9/dist-packages/PIL/JpegImagePlugin.py", line 621, in get_sampling
    sampling = im.layer[0][1:3] + im.layer[1][1:3] + im.layer[2][1:3]
  File "/usr/local/lib/python3.9/dist-packages/PIL/Image.py", line 517, in __getattr__
    raise AttributeError(name)
AttributeError: layer
```

### Operating system
OSx, Ubuntu, etc

### Your thumbor.conf
Original thumbor.conf

### PR DESCRIPTION
**Pull request by RaphaelVRossi**
Only call get_sampling on images with layers and layer.

Closes #1486

### CODE REVIEW

**Review by scorphus — CHANGES_REQUESTED** [review]
Good catch!

I'd love to see a test case for that 😊

Would you kindly do that? 🤗

**Comment by RaphaelVRossi** [comment]
> Good catch!
> 
> I'd love to see a test case for that 😊
> 
> Would you kindly do that? 🤗

Hey @scorphus, created PSD unit test, and validate `engine.subsampling` on JPG, JPEG2000 and PSD files.

File: thumbor/engines/pil.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def create_image(self, buffer):
         self.exif = img.info.get("exif")
         self.original_mode = img.mode
 
-        if getattr(img, "layers", None):
+        if getattr(img, "layers", None) and getattr(img, "layer", None):
```
Comment (scorphus) [inline thumbor/engines/pil.py:84]: Differently form `layers`, which can be `0` and therefore rules out sub-sampling ([PIL/Jpeg2KImagePlugin.py](https://github.com/python-pillow/Pillow/blob/243402e78e2bf2b7af478c6d891816e372b5c3f9/src/PIL/Jpeg2KImagePlugin.py#L226)):
```py
        self.layers = 0
```

`layer` is either defined or not ([PIL/JpegImagePlugin.py](https://github.com/python-pillow/Pillow/blob/243402e78e2bf2b7af478c6d891816e372b5c3f9/src/PIL/JpegImagePlugin.py#L354-L358)):
```py
        # Create attributes
        self.bits = self.layers = 0


        # JPEG specifics (internal)
        self.layer = []
```

So, this line can be replaced:

```suggestion
        if hasattr(img, "layer"):
```

**Review by scorphus — CHANGES_REQUESTED** [review]
Thanks for writing tests, @RaphaelVRossi! 👍 

I recommend a few changes below. Please consider committing the suggestion, we can squash the commits later, when merging 😉

File: tests/engines/test_pil.py [hunk 2]
```diff
@@ -59,13 +59,25 @@ def test_load_image(self):
...
         with open(join(STORAGE_PATH, "image.jp2"), "rb") as image_file:
             buffer = image_file.read()
         image = engine.create_image(buffer)
         expect(image.format).to_equal("JPEG2000")
+        expect(engine.subsampling).to_be_null()
+
+    def test_load_psd_image(self):
+        engine = Engine(self.context)
+        with open(
+            join(STORAGE_PATH, "Guido-portrait-2014.psd"), "rb"
```
[the remark this answers is not in the document]
Comment (RaphaelVRossi) [reply tests/engines/test_pil.py]: Thanks for this suggestion! 

File changed.

File: thumbor/engines/pil.py (hunk 1 above)
Comment (RaphaelVRossi) [reply thumbor/engines/pil.py]: Done!

**Review by scorphus — APPROVED** [review]
Thanks, @RaphaelVRossi! :shipit:

**Review by devppjr — APPROVED** [review]
LGTM!! 🚀

**Pull request merged**